File: generate_embeddings.py
import openai
import json
import os

openai.api_key = os.getenv("OPENAI_API_KEY")

entries = [
    {
        "question": "when did the chest pain start",
        "match": "About 10 minutes ago while sitting at the park."
    },
    {
        "question": "do you take nitro",
        "match": "Yes, I have nitro and take it if I feel pressure in my chest."
    },
    {
        "question": "do you feel shortness of breath",
        "match": "Yes, a little, especially if I try to walk."
    },
    {
        "question": "are you allergic to anything",
        "match": "No, I have no known allergies."
    }
]

output = []

for entry in entries:
    print(f"Embedding: {entry['question']}")
    response = openai.Embedding.create(
        input=entry["question"],
        model="text-embedding-3-small"
    )
    embedding = response['data'][0]['embedding']
    output.append({
        "question": entry["question"],
        "embedding": embedding,
        "match": entry["match"]
    })

with open("embeddings.json", "w") as f:
    json.dump(output, f, indent=2)

print("✅ embeddings.json generated.")

File: generate_embeddings.js
// Run this locally with: node generate_embeddings.js
const fs = require('fs');
const { Configuration, OpenAIApi } = require('openai');

const configuration = new Configuration({
  apiKey: process.env.OPENAI_API_KEY
});
const openai = new OpenAIApi(configuration);

const questions = [
  {
    question: "do you have chest pain?",
    answer: "Yes, it's a heavy pressure right here in my chest.",
    role: "patient"
  },
  {
    question: "what is the blood pressure",
    answer: "Blood pressure is 130/84 mmHg.",
    role: "proctor"
  },
  {
    question: "how many patients",
    answer: "You see only one patient.",
    role: "proctor"
  }
];

async function generateEmbeddings() {
  for (const item of questions) {
    const response = await openai.createEmbedding({
      model: "text-embedding-ada-002",
      input: item.question
    });
    item.embedding = response.data.data[0].embedding;
  }

  fs.writeFileSync('vector-db.json', JSON.stringify(questions, null, 2));
  console.log("✅ vector-db.json created with embeddings.");
}

generateEmbeddings();
